### DISCUSSION TITLE
Use custom fallback node when custom node name not found in nodeTypes

### DISCUSSION THREAD

**Discussion by spagrhetti** [xyflow/xyflow#4407] [Q&A] [answered]
I am using custom node types like this:

```
const nodeTypes = {"customName": CustomNameNode};

<ReactFlow
   nodeTypes={nodeTypes}
...
/>
```

However, I want all names that are NOT in this nodeTypes dict to all fall back to a custom node type. So instead of getting this:

```
Node type "customName2" not found. Using fallback type "default".
```

I would like to have it all fall back to some custom fallback node that I define. Is this possible? I tried doing a proxy like this:

```
const nodeTypesHandler = {
  get: function (target, name) {
    return name in target ? target[name] : MyCustomFallbackNode;
  },
};

const nodeTypesProxy = new Proxy(nodeTypes, nodeTypesHandler);
```

But that ended up in some very very weird behavior (my ReactFlow was very laggy and the nodes had weird extra borders around them).

**Comment by moklick** [comment, marked as answer]
You can overwrite the default one: 

```js
const nodeTypes = {
  customName: CustomNameNode,
  default: DefaultCustomNode
};
```

You would still get the warning while developing but the correct node would be shown.

**Reply by spagrhetti** [reply]
Wow! Didn't know you could do this, thank you so much!
